segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/IndexRequestProcessor.java

```diff
@@ -81,8 +82,8 @@ private static Map.Entry<Long, Long> applySearch(StreamSegmentStore store, Strin
         SegmentProperties properties = store.getStreamSegmentInfo(indexSegmentName, TIMEOUT).join();
         long startIdx = properties.getStartOffset() / ENTRY_SIZE;
         long endIdx = properties.getLength() / ENTRY_SIZE;
-
-        return SortUtils.newtonianSearch(idx -> {
+        //If startIdx and endIdx are same, then pass length of segment as a result.
+        return startIdx == endIdx ? getSegmentLength(store, segment, startIdx) : SortUtils.newtonianSearch(idx -> {
```

Comment:
As this is a return you can break out the turanary into an if/return and not have any further indentation.

Reply:
Done.